client/my-sites/email/email-providers-stacked-comparison/provider-cards/utils.tsx

```diff
@@ -0,0 +1,149 @@
+import formatCurrency from '@automattic/format-currency';
+import { RequestCartProduct } from '@automattic/shopping-cart';
+import { translate } from 'i18n-calypso';
+import page from 'page';
+import React, { FunctionComponent, ReactElement } from 'react';
+import PromoCardPrice from 'calypso/components/promo-section/promo-card/price';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { fillInSingleCartItemAttributes } from 'calypso/lib/cart-values';
+
+export enum TermLength {
+	ANNUALLY = 'annually',
+	MONTHLY = 'monthly',
+}
+
+export const formattedPriceWithTerm = (
+	formattedPriceClassName: string,
+	termLengthClassName: string,
+	cost: number,
+	currencyCode: string,
+	termLength?: TermLength
+) => {
+	return translate( '{{price/}} /mailbox {{termLength/}}', {
+		components: {
+			price: (
+				<span className={ formattedPriceClassName }>
+					{ formatCurrency( cost ?? 0, currencyCode ) }
+				</span>
+			),
+			termLength: <span className={ termLengthClassName }>/{ termLength }</span>,
+		},
+		comment:
+			'{{price/}} is the formatted price, e.g. $20' +
+			'{{termLength/}} is already translated and it is either annually or monthly',
+	} );
+};
+
+export const AddToCartAndCheckout = (
+	shoppingCartManager: any,
+	cartItem: RequestCartProduct,
+	productList: any,
+	setAddingToCart: ( addingToCart: boolean ) => void,
+	selectedSite: string
+): void => {
+	setAddingToCart( true );
+
+	shoppingCartManager
+		.addProductsToCart( [ fillInSingleCartItemAttributes( cartItem, productList ) ] )
+		.then( ( response: any ) => {
+			setAddingToCart( false );
+			const { errors } = response?.messages;
```

Comment:
In https://github.com/Automattic/wp-calypso/pull/58834 I'm working on making cart action promises reject if there are errors. I suggest that you add a `.catch()` block here that does `setAddingToCart(false)` so you're future-proof against that case.